changebot/github/github_api.py

```diff
@@ -99,6 +103,51 @@ def get_file_contents(self, path_to_file):
         contents_base64 = response.json()['content']
         return base64.b64decode(contents_base64).decode()
 
+    def get_user_config(self, path_to_file='.astropybot.yml',
+                        warn_on_failure=True):
+        """
+        Load user configuration for bot.
+
+        Parameters
+        ----------
+        path_to_file : str
+            Configuration file in YAML format in the repository.
+            If not given or invalid, default is used.
+
+        warn_on_failure : bool
+            Emit warning on failure to load YAML file.
+
+        Returns
+        -------
+        cfg : dict
+            Configuration parameters.
+
+        """
+        try:
+            file_content = self.get_file_contents(path_to_file)
+            cfg = yaml.load(file_content)
+        except Exception as e:
+            if warn_on_failure:
+                warnings.warn(str(e))
+            # Empty dict means calling code set the default
+            cfg = {}
+
+        return cfg
+
+    def get_config_value(self, cfg_key, cfg_default):
+        """
+        Convenience method to extract user config from global cache.
+        """
+        global cfg_cache
+
+        cfg_cache_key = (self.repo, self.branch, self.installation)
+        if cfg_cache_key not in cfg_cache:
+            cfg_cache[cfg_cache_key] = self.get_user_config(
+                warn_on_failure=False)
```

Comment:
NOTE: I turn off warning here mainly to make `pytest` happy but I figure it is not needed much anyway except when there is a need for debugging.